### ISSUE TITLE
Launchers not installing 

### ISSUE THREAD

**Issue by Bartell17** [moraroy/NonSteamLaunchers-On-Steam-Deck#296]
No launchers are installing. Discussed on Reddit with @moraroy 

[NonSteamLaunchers-install.log](https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/files/15349782/NonSteamLaunchers-install.log)

Attached latest install log file for Epic (was set away to install via Decky)

**Comment by Bartell17** [comment]
> Same here, I've tried Battle.net. Did a fresh start too, it updates Battle.net and then crashes. Steam restarts itself, but no Battle.net installed. From the logs I can see that some Python module is missing:
> 
> [NonSteamLaunchers-install.log](https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/files/15354279/NonSteamLaunchers-install.log)
> 
> Installed from the .desktop file, SteamOS 3.6.3, latest preview build.

Oh 3.6.3 is a preview build? It's showing in my SD and constantly tries to install and doesn't. It's been annoying me thinking my deck was faulty. Why will it present a preview build if I'm on stable in settings menu? Weird

**Comment by moraroy** [comment]
@Pipodi @Bartell17 these are new errors, i never needed these python modules before, so im assuming it has to do with steams new updates/ or the modules need updating thank you for letting me know will try to find a fix!

**Comment by tmespe** [comment]
I fixed this on my end by downloading charset_normalizer from their repo and adding it to the modules folder.

**Comment by moraroy** [comment]
@tmespe gotcha, but why is it needed now all of a sudden?

**Comment by tmespe** [comment]
@moraroy My guess would be that Valve removed either chardet and/or charset_normalizer from being installed by default on 3.6. They are not currently installed on 3.6.3 at least. I will revert back to stable and check if they are there later today.

**Comment by Notaexpert** [comment]
as of yet is there a solution?

**Comment by Notaexpert** [comment]
Im trying to run ps plus and its not installing

**Comment by moraroy** [comment]
@Notaexpert not yet, perhaps in due time

**Comment by Notaexpert** [comment]
thanks for the response

**Comment by tmespe** [comment]
@moraroy 
I just reimaged to the Stable channel on the deck and I'm seeing both python-chardet (5.1.0-3) and python-charset-normalizer (3.1.0-3) installed by default with Pacman -Q. Not sure why valve removed them from the 3.6 image.

For anyone looking for a workaround you can [download zip of the charset_normalizer repo](charset_normalizer ), extract the zip file, copy the charset_normalizer folder into /home/.config/systemd/user/Modules.

**Comment by moraroy** [comment]
@tmespe makes sense why I was not able to get this error because I had also recently reset my deck. Good to know hopefully the next steam update will fix everyone's deck from that error.

**Comment by Notaexpert** [comment]
> @moraroy I just reimaged to the Stable channel on the deck and I'm seeing both python-chardet (5.1.0-3) and python-charset-normalizer (3.1.0-3) installed by default with Pacman -Q. Not sure why valve removed them from the 3.6.3 image.
> 
> For anyone looking for a workaround you can [download zip of the charset_normalizer repo](charset_normalizer), extract the zip file, copy the charset_normalizer folder into /home/.config/systemd/user/Modules.

could you please explain in simpler terms

**Comment by mac-roj** [comment]
Hi there, i have experienced the same issue on a fresh new steam deck, my SteamOS is 3.5.19.
Install log attached below.
[NonSteamLaunchers-install.log](https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/files/15375638/NonSteamLaunchers-install.log)

**Comment by lyledouglass** [comment]
I'm not sure this is only 3.6.3 - I recently got a steamdeck and am on 3.5.19 image and am having the same issue. I also did a factory reset just to be sure my issue is the same. I'll upload my log right after this comment

**Comment by lyledouglass** [comment]
> I'm not sure this is only 3.6.3 - I recently got a steamdeck and am on 3.5.19 image and am having the same issue. I also did a factory reset just to be sure my issue is the same. I'll upload my log right after this comment

[NonSteamLaunchers-install.log](https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/files/15382040/NonSteamLaunchers-install.log)

**Comment by moraroy** [comment]
@lyledouglass add a random non steam shortcut to your library first...then try nonsteamlaunchers again.

**Comment by lyledouglass** [comment]
That worked, not the same issue it seems! Thanks

**Comment by willymai** [comment]
i install battle net and it's stucked at 99% using NonSteamLauncher 3.8.9 and 3.8.8 have same issue

**Comment by GeorgyDesign** [comment]
Hey, folks!

Is the issue solved? I just got Steam Deck OLED yesterday, so gonna update it and whatnot — wanted to use @moraroy solution for other stores, but decided to check the issues since there where not any updates for quite some time. 

Does it work? Did new Steam update fix the issue? 

Also, my apologies for asking it here, but can I install be launchers after previous ones? E.g., I installed EGS and decided to install Ubisoft Connect next month? How to do it?

**Comment by moraroy** [comment]
@GeorgyDesign I would just use the plug-in from deck loader if you're starting on a fresh deck. Go to the deck loader testing store and download nonsteamlaunchers from there.

**Comment by KhazP** [comment]
> @GeorgyDesign I would just use the plug-in from deck loader if you're starting on a fresh deck. Go to the deck loader testing store and download nonsteamlaunchers from there.

That doesn't work for me either. I'm on the stable branch in SD, and after I install EA and UPlay, even though it says successfully installed, they don't appear. Funny enough only Rockstar Launcher appears among those three.

**Comment by joshuataylor** [comment]
It seems the dependency as noted isn't installed on preview, and there might be two seperate issues occuring here:

1. You need a nonsteam game added already
2. The dependency isn't installed on preview.

To fix 2:

1. Enable r/w mode on the deck.
2. sudo pacman --Sy python-chardet python-charset-normalizer
3. Rerun the script

It seems to fix the issue, at least with battlenet.

**Comment by KhazP** [comment]
> It seems the dependency as noted isn't installed on preview, and there might be two seperate issues occuring here:
> 
> 1. You need a nonsteam game added already
> 2. The dependency isn't installed on preview.
> 
> To fix 2:
> 
> 1. Enable r/w mode on the deck.
> 2. sudo pacman --Sy python-chardet python-charset-normalizer
> 3. Rerun the script
> 
> It seems to fix the issue, at least with battlenet.

Actually the latest update fixed it :)

**Comment by joshuataylor** [comment]
I tried the latest one today, 2024-06-12) and could not see Battlenet.

**Comment by mbrow237** [comment]
I'm using the latest version of the script and tried installing the EA App and ItchIO. EA app shows up in my library but not ItchIO.

I tried multiple times with multiple 'start fresh' and ItchIO is still not showing up.

**Comment by moraroy** [comment]
Can you show me your log?

**Comment by mbrow237** [comment]
> Can you show me your log?

Sorry for the delay. I am attaching my log. I went ahead and remove information around my steam users and my steam shortcuts(I have over 1500 non steam games installed).
[NonSteamLaunchers-install.log](https://github.com/user-attachments/files/15889941/NonSteamLaunchers-install.log)

**Comment by moraroy** [comment]
@mbrow237 this has been fixed just now in the desktop version , unfortunately i can no longer hotfix things to the decky plugin....so all updates will be slower to the plugin

**Comment by mbrow237** [comment]
Oh I am having this issue with the desktop version as well, I tried the decky version last to see if it worked there. I will re-download the desktop version and provide the logs shortly.

**Comment by moraroy** [comment]
@mbrow237 it should work just tested on mine

**Comment by mbrow237** [comment]
Ok I did a fresh start, Deleted the program, and downloaded the the desktop version of v3.8.9. This time, I selected Battle.net(first time install) and Itchio. After it finished installing, I can see Battle.net in steam, but not Itchio. I attached my log
[NonSteamLaunchers-install-copy.log](https://github.com/user-attachments/files/15896428/NonSteamLaunchers-install-copy.log)

**Comment by moraroy** [comment]
@mbrow237 try to install itchip by itself?

**Comment by moraroy** [comment]
@mbrow237 for some reason, battle.net is actually breaking the whole script when it finishes installing so it skips all the other launchers that you selected lol I don't know why

**Comment by moraroy** [comment]
@mbrow237 just added another update to the desktop file. i was able to fix it totally on my deck, should work now!

**Comment by mbrow237** [comment]
Ok so I gave it another go, but something weird happened.

I did another fresh start, and then I only try installing just ItchIO. After it finished installing and steam restarted, all my non steam game shortcuts were gone. 

I was able to get my non steam games back by using the shortcut-backup.vdf file. Funny enough, now the ItchIO launcher is appearing in my steam lol.

**Comment by moraroy** [comment]
@mbrow237 dang that's not good, somehow it's corrupting the vdf file......I don't think I changed any code.....have you always used the desktop version of NSL? What about the olugin?

**Comment by moraroy** [comment]
@mbrow237 I just added a small fix, let me.know if you can recreate that bug?

**Comment by Jzooor** [comment]
@moraroy I've just tried using this for the first time to install the EA app. I appears to have installed something, but nothing was added to steam.
[NonSteamLaunchers-install.log](https://github.com/user-attachments/files/15906795/NonSteamLaunchers-install.log)

**Comment by moraroy** [comment]
@Jzooor please try again, open up nsl and click on NSLGamescanner and select yes if it ask you to restart this will grab the fix i just pushed just now let me know if it works!

**Comment by Jzooor** [comment]
@moraroy it worked. Thanks!

**Comment by moraroy** [comment]
@Jzooor should be fixed in the latest update?
